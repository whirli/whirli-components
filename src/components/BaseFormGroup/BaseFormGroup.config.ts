// @ts-ignore
import type { PropType } from 'vue';

import { ComponentStyles, ComponentProps } from '../../@types/components';

// Local config
import config from '@whirli-local/components/BaseFormGroup/BaseFormGroup.config';

export const ConfigStyles: ComponentStyles = {
  ...config.styles,
};

export const ConfigProps: ComponentProps = {
  ...config.props,
};
