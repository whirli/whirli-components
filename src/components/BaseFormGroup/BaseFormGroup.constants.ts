export const PropKeys: Record<string, string> = {
  ERROR: 'error',
  NAME: 'name',
  LABEL: 'label',
};

export const PropValues: Record<string, string[]> = {};
