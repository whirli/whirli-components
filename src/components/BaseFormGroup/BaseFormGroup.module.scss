@import '@whirli-components/styles/Breakpoints.module.scss';

// default styles
.form-group {}
// breakpoint styles
$packageBreakpointStyles: ();