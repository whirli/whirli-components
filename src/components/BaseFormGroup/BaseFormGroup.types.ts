import { PropKeys, PropValues } from './BaseFormGroup.constants';

const propDirection: string[] = PropValues[PropKeys.DIRECTION];

export type PropDirection = typeof propDirection[number];
