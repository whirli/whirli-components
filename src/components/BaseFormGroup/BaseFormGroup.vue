<template>
  <div :class="classes"><slot /></div>
</template>

<script setup lang="ts">
// Vue
;

// Styles
import styles from '@whirli-local/components/BaseFormGroup/BaseFormGroup.module.scss';

// Data
import { ConfigStyles, ConfigProps } from './BaseFormGroup.config';

// Types
import { ComponentStyles as ComponentStylesInterface } from '@whirli-components/@types/components';
import { Props } from '@whirli-components/@types/props';

const ComponentStyles: ComponentStylesInterface = ConfigStyles;

const props: Props = defineProps(ConfigProps);

// Classes
import useClasses from '@whirli-components/@use/class';
const { makeClasses } = useClasses();
const classes: string[] = [styles['form-group'], ...makeClasses(ComponentStyles, props, styles)];
</script>
