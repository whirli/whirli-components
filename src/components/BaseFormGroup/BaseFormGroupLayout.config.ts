// @ts-ignore
import type { PropType } from 'vue';

import { ComponentStyles, ComponentProps } from '@whirli-components/@types/components';

// Local config
import config from '@whirli-local/components/BaseFormGroup/BaseFormGroup.config';
import { PropKeys } from './BaseFormGroup.constants';

export const ConfigStyles: ComponentStyles = {
  ...config.styles,
};

export const ConfigProps: ComponentProps = {
  [PropKeys.NAME]: {
    required: true,
    type: String as PropType<string>,
  },
  [PropKeys.LABEL]: {
    required: true,
    type: String as PropType<string>,
  },
  [PropKeys.ERROR]: {
    required: false,
    type: String as PropType<string>,
    default: null,
  },
  ...config.props,
};
