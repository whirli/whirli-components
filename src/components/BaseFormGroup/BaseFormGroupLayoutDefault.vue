<template>
  <BaseFormGroup>
    <BaseGrid gap="0">
      <BaseRow>
        <BaseFormLabel :name="props.name">{{ props.label }}</BaseFormLabel>
      </BaseRow>
      <BaseRow>
        <slot />
      </BaseRow>
      <BaseRow v-if="props.error">
        <BaseFormError>{{ props.error }}</BaseFormError>
      </BaseRow>
    </BaseGrid>
  </BaseFormGroup>
</template>

<script setup lang="ts">
// Data
import { ConfigProps } from './BaseFormGroupLayout.config';

// Types
import { Props } from '@whirli-components/@types/props';

// @ts-ignore
const props: Props = defineProps(ConfigProps);

// Components
import BaseFormGroup from '@whirli-components/components/BaseFormGroup/BaseFormGroup.vue';
import BaseGrid from '@whirli-components/components/BaseGrid/BaseGrid.vue';
import BaseRow from '@whirli-components/components/BaseRow/BaseRow.vue';
import BaseFormLabel from '@whirli-components/components/BaseFormLabel/BaseFormLabel.vue';
import BaseFormError from '@whirli-components/components/BaseFormError/BaseFormError.vue';
</script>
