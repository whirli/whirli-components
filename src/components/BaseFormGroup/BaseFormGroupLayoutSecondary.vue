<template>
  <BaseFormGroup>
    <BaseGrid gap="0">
      <BaseRow :direction="rowDirection">
        <BaseFormLabel :name="props.name">{{ props.label }}</BaseFormLabel> <slot />
      </BaseRow>
      <BaseRow v-if="props.error">
        <BaseFormError>{{ props.error }}</BaseFormError>
      </BaseRow>
    </BaseGrid>
  </BaseFormGroup>
</template>

<script setup lang="ts">
// Data
import { ConfigProps as SharedConfigProps } from './BaseFormGroupLayout.config';
import { ConfigProps } from './BaseFormGroupLayoutSecondary.config';

// Types
import { Props } from '@whirli-components/@types/props';
import { PropDirection } from './BaseFormGroup.types';

// @ts-ignore
const props: Props = defineProps({ ...SharedConfigProps, ...ConfigProps });

// Constants
import { PropKeys } from './BaseFormGroup.constants';

// Components
import BaseFormGroup from '@whirli-components/components/BaseFormGroup/BaseFormGroup.vue';
import BaseGrid from '@whirli-components/components/BaseGrid/BaseGrid.vue';
import BaseRow from '@whirli-components/components/BaseRow/BaseRow.vue';
import BaseFormLabel from '@whirli-components/components/BaseFormLabel/BaseFormLabel.vue';
import BaseFormError from '@whirli-components/components/BaseFormError/BaseFormError.vue';

const rowDirection: PropDirection = props[PropKeys.DIRECTION] === 'reverse' ? 'row-reverse' : 'row';
</script>
