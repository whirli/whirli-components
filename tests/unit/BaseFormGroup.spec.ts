import { shallowMount } from '@vue/test-utils';
import BaseFormGroup from '@/components/BaseFormGroup/BaseFormGroup.vue';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
let wrapper: any;
const text = 'Hello World';

describe('BaseFormGroup.vue', () => {
  beforeEach(() => {
    wrapper = shallowMount(BaseFormGroup, {
      slots: {
        default: text,
      },
    });
  });

  it('renders the component', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('displays the correct text', () => {
    expect(wrapper.text()).toMatch(text);
  });
});
