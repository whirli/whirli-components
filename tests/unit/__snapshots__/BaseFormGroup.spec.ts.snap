// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseFormGroup.vue renders the component 1`] = `
<div
  class=""
>
  
  Hello World
  
</div>
`;
